envi/archs/i386/emu.py

```diff
@@ -272,18 +272,21 @@ def doRepzPrefix(emu, meth, op):
         ZF starts off being set. 
         Then the instruction is repeated and ECX decremented until either
         ECX reaches 0 or the ZF is cleared.
+        
+        Respects emu option "i386:repmax" limiting the number of repetitions.
         '''
         ecx = emu.getRegister(REG_ECX)
         emu.setFlag(EFLAGS_ZF, 1)
 
         repmax = emu.getEmuOpt('i386:repmax')
         if repmax:
-            ecx = min(ecx, repmax)
+            repmax = min(ecx, repmax)
 
         ret = None
-        while ecx and emu.getFlag(EFLAGS_ZF):
+        while ecx and repmap and emu.getFlag(EFLAGS_ZF):
```

Comment:
small typo in `repmap` causing some problems.

Reply:
```suggestion
        while ecx and repmax and emu.getFlag(EFLAGS_ZF):
```